beetsplug/kodinfo.py

```diff
@@ -0,0 +1,495 @@
+                    album.path.decode('utf8'),
+                    os.pardir, 'artist.tbn')
+                urllib.request.urlretrieve(
+                    artist_info(albumid)[11], thumb_location)
+                thumb = et.SubElement(root, 'thumb')
+                thumb.text = artist_info(albumid)[11]
+                for i in range(len(thumbs('artist', albumid))):
+                    thumb = et.SubElement(root, 'thumb')
+                    thumb.text = thumbs('artist', albumid)[i]
+
+            fanart = et.SubElement(root, 'fanart')
+            fanart.text = artist_info(albumid)[12]
+
+            for i in range(len(artist_albums(artistid))):
+                album = et.SubElement(root, 'album')
+                title = et.SubElement(album, 'title')
+                title.text = artist_albums(artistid)[i][1]
+                year = et.SubElement(album, 'year')
+                year.text = str(artist_albums(artistid)[i][0])
+
+            xml = et.tostring(
+                root,
+                pretty_print=True,
+                xml_declaration=True,
+                encoding='UTF-8',
+                standalone="yes").decode()
+            print(xml, file=open(artistnfo, 'w+'))
+
+
+def update_kodi(host, port, user, password):
+    """Send request to the Kodi api to start a library refresh."""
+    """By Pauli Kettunen"""
+    url = "http://{0}:{1}/jsonrpc/".format(host, port)
+
+    """Content-Type: application/json is mandatory"""
+    """according to the kodi jsonrpc documentation"""
+
+    headers = {'Content-Type': 'application/json'}
+
+    # Create the payload. Id seems to be mandatory.
+    payload = {'jsonrpc': '2.0', 'method': 'AudioLibrary.Scan', 'id': 1}
+    r = requests.post(
+        url,
+        auth=(user, password),
+        json=payload,
+        headers=headers)
+
+    return r
+
+
+class KodiNfo(BeetsPlugin):
+    """KodiNfo Plugin."""
+
+    def __init__(self):
+        """Plugin docstring."""
+        super(KodiNfo, self).__init__()
+
+        # Adding defaults.
+        self.config['kodi'].add({
+            u'host': u'localhost',
+            u'port': 8080,
+            u'user': u'kodi',
+            u'pwd': u'kodi',
+            u'nfo_format': 'xml'})
+        config['kodi']['pwd'].redact = True
+        self.register_listener('album_imported', self.create_nfos)
+        self.register_listener('database_change', self.listen_for_db_change)
+
+    def create_nfos(self, lib, album):
+        """Create nfos as per choice in config."""
+        try:
+            # Check if MBID is valid UUID as per MB recommendations
+            UUID(album.mb_albumid)
+            self._log.info(u'Album ID is valid MBID...creating .nfos')
+            albumid = 'mb_albumid:' + album.mb_albumid
+            mb_albumid = album.mb_albumid
+            mb_artistid = album.mb_albumartistid
+            nfo_format = '{0}'.format(config['kodi']['nfo_format'])
+            if nfo_format in 'mbid_only_text':
```

Comment:
I recommend a shorter name like `text` for this configuration option—it could be a little less confusing.